.travis.yml

```diff
@@ -1,32 +1,106 @@
-  - docker-compose up -d db
-  - docker-compose up -d redis
-  - sleep 10
-  - docker-compose run web rake db:create
-install:
-  - gem install brakeman
-script:
-  - docker-compose run --rm webtest "./node_modules/.bin/eslint app/assets/src --ext .js,.jsx"
-  - echo "quit" | docker-compose run --rm webtest "rails dbconsole"
-  - docker-compose run --rm webtest "rubocop -R --config .rubocop_todo.yml"
-  - docker-compose run --rm webtest "rake db:drop db:create db:migrate db:seed"
-  - docker-compose run --rm webtest "rails test"
-  - docker-compose run --rm webtest "rake db:drop db:create db:migrate db:seed"
-  - docker-compose run --rm webtest "rspec"
-  - docker-compose run --rm webtest "npm test"
-  - docker-compose run --rm webtest "pip install mock; python -m unittest discover -v -s /app/test/ -p 'test_*.py'"
-  # Block on High Confidence warnings
-  - brakeman --no-pager -w3
+jobs:
+  include:
+    - name: "Rails Test"
+      language: ruby
+      rvm: 2.5
+      cache:
+        bundler: true
+        directories:
+          - $HOME/.cache/pip
+      env:
+        # foo
+        - DB_USERNAME: travis
+        - DB_PASSWORD: root
+        - SAMPLES_BUCKET_NAME: idseq-samples-development
+      services:
+        - mysql
+        - redis
+      before_install:
+        # Emulate docker compose routing
+        #   Docker compose maps the host db and redis to the mysql and redis containers
+        #   We are using travis-ci services, which use localhost, for testing instead of docker compose 
+        #   we need to map these hosts to localhost to maintain compatibility between the two.
+        - sudo bash -c "echo '127.0.0.1 db' >> /etc/hosts" 
+        - sudo bash -c "echo '127.0.0.1 redis' >> /etc/hosts" 
+        - mkdir -p ~/.aws && printf "[default]\nregion = $AWS_REGION\naws_access_key_id = $AWS_ACCESS_KEY_ID\naws_secret_access_key = $AWS_SECRET_ACCESS_KEY\n" > ~/.aws/credentials
+        # Install python and dependencies
+        - source ~/virtualenv/python2.7/bin/activate
+        - pip install -r requirements.txt --ignore-installed PyYAML
+      before_script:
+        # Create dummy versions of our javascript and css bundle
+        #   The actual files are not required for testing ruby, just the presence of the files
+        - mkdir -p app/assets/dist
+        - echo foo > app/assets/dist/bundle.min.css
+        - echo foo > app/assets/dist/bundle.min.js
+        - echo foo > app/assets/dist/main.bundle.min.css
+        - echo foo > app/assets/dist/main.bundle.min.css.map
+        - echo foo > app/assets/dist/main.bundle.min.js
+        - echo foo > app/assets/dist/main.bundle.min.js.map
+        - echo foo > app/assets/dist/vendors~main.bundle.min.css
+        - echo foo > app/assets/dist/vendors~main.bundle.min.css.map
+        - echo foo > app/assets/dist/vendors~main.bundle.min.js
+        - echo foo > app/assets/dist/vendors~main.bundle.min.js.map
```

Comment:
Can this be made less manual? e.g. somehow run bundler here to create an empty bundle?

Reply:
I will research if webpack can support this